### PR TITLE
Add style support for navigation bar.

### PR DESCRIPTION
**Pull request by k0nserv**
I stupidly forgot to do this previously 😐.

### CODE REVIEW

File: changelog.md [hunk 1]
```diff
@@ -7,6 +7,7 @@
 - react-native-bpk-component-navigation-bar:
   - `navigationBarStatusBarStyle` is now not supported on iOS.
   - New chevron style for `BpkNavigationBarBackButtonIOS` that better alings with iOS.
+  - Support use provided `style` prop.
```
Comment (shaundon) [inline changelog.md]: How about `Support for using the provided style prop`?

File: native/packages/react-native-bpk-component-navigation-bar/src/BpkNavigationBar-test.android.js [hunk 2]
```diff
@@ -73,6 +73,25 @@ describe('android', () => {
       expect(tree).toMatchSnapshot();
     });
 
+    it('should render correctly', () => {
```
Comment (shaundon) [inline native/packages/react-native-bpk-component-navigation-bar/src/BpkNavigationBar-test.android.js]: How about `'should render correctly with custom styles'` so as not to duplicate the name from above?

File: native/packages/react-native-bpk-component-navigation-bar/src/BpkNavigationBar.android.js [hunk 3]
```diff
@@ -109,13 +111,17 @@ class BpkNavigationBar extends Component<Props, {}> {
     leadingButton: PropTypes.element,
     trailingButton: PropTypes.element,
     subtitleView: PropTypes.element,
+
+    // FIXME: We need a better prop type for style
+    style: PropTypes.object, // eslint-disable-line react/forbid-prop-types
```
Comment (shaundon) [inline native/packages/react-native-bpk-component-navigation-bar/src/BpkNavigationBar.android.js]: Can't we use `ViewPropTypes.style` here?

File: native/packages/react-native-bpk-component-navigation-bar/src/BpkNavigationBar.ios.js [hunk 4]
```diff
@@ -106,6 +106,7 @@ export type Props = {
   leadingButton: ?Element<ButtonType>,
   trailingButton: ?Element<ButtonType>,
   subtitleView: ?Element<any>,
+  style: any,
```
Comment (shaundon) [inline native/packages/react-native-bpk-component-navigation-bar/src/BpkNavigationBar.ios.js]: Shouldn't this be optional? Or does `any` encapsulate that

File: native/packages/react-native-bpk-component-navigation-bar/src/BpkNavigationBar.android.js [hunk 5]
```diff
@@ -97,6 +98,7 @@ export type Props = {
   leadingButton: ?Element<BpkNavigationBarButtonAndroid>,
   trailingButton: ?Element<BpkNavigationBarButtonAndroid>,
   subtitleView: ?Element<any>,
+  style: any,
```
Comment (matteo-hertel) [inline native/packages/react-native-bpk-component-navigation-bar/src/BpkNavigationBar.android.js]: Actually, sorry the type here should be `?(Object | Array<Object>)`

like 
https://github.com/Skyscanner/backpack/blob/c70f45082ff58d0d8971dd3d451fdc151bf44a18/native/packages/react-native-bpk-component-button-link/src/common-types.js#L39
https://github.com/Skyscanner/backpack/blob/3893710c9d7ba1957bf71bc7343a4bf1a79c8a67/native/packages/react-native-bpk-component-phone-input/src/BpkFlag.js#L31

File: native/packages/react-native-bpk-component-navigation-bar/src/BpkNavigationBar.ios.js (hunk 4 above)
Comment (matteo-hertel) [inline native/packages/react-native-bpk-component-navigation-bar/src/BpkNavigationBar.ios.js]: same as above

File: native/packages/react-native-bpk-component-navigation-bar/src/BpkNavigationBar.android.js (hunk 5 above)
Comment (k0nserv) [reply native/packages/react-native-bpk-component-navigation-bar/src/BpkNavigationBar.android.js]: Didn't we have issues with that specifically? cc @matthewdavidson

**Pull request merged**